azure/azure_infra/cpd-install.tf

```diff
@@ -582,3 +584,96 @@ resource "null_resource" "install_cpd_spss" {
+        ]
+    }
+    depends_on = [
+        null_resource.install_cpd_lite,
+        null_resource.install_cpd_dv,
+        null_resource.install_cpd_openscale,
+        null_resource.install_cpd_wsl,
+        null_resource.install_cpd_wkc,
+        null_resource.install_cpd_wml,
+        null_resource.install_cpd_spark,
+        null_resource.install_cpd_cde,
+        null_resource.install_cpd_streams,
+        null_resource.install_cpd_streams_flows,
+        null_resource.install_cpd_ds,
+        null_resource.install_cpd_db2wh,
+        null_resource.install_cpd_db2oltp,
+        null_resource.install_cpd_dods,
+        null_resource.install_cpd_ca,
+        null_resource.install_cpd_spss
+    ]
+}
+
+resource "null_resource" "install_cpd_watson_discovery" {
+    count = var.watson-discovery == "yes" && var.accept-cpd-license == "accept" ? 1 : 0
+    triggers = {
+        bootnode_ip_address = azurerm_public_ip.bootnode.ip_address
+        username = var.admin-username
+        private_key_file_path = var.ssh-private-key-file-path
+        namespace = var.cpd-namespace
+    }
+    connection {
+        type = "ssh"
+        host = azurerm_public_ip.bootnode.ip_address
+        user = var.admin-username
+        private_key = file(self.triggers.private_key_file_path)
+    }
+    provisioner "remote-exec" {
+        inline = [
+            "cat > ${local.installerhome}/elasticsearch-mc.yaml <<EOL\n${file("../openshift_module/elasticsearch-machineconfig.yaml")}\nEOL",
+            "oc create -f ${local.installerhome}/elasticsearch-mc.yaml",
+            "cat > ${local.installerhome}/watson-discovery-override.yaml <<EOL\n${data.template_file.watson-discovery-override.rendered}\nEOL",
+            "docker_secret=$(oc get secrets | grep default-dockercfg | awk '{print $1}')",
+            "sed -i s/default-dockercfg-xxxxx/$docker_secret/g ${local.installerhome}/watson-discovery-override.yaml",
+            "image_registry_route=$(oc get route -n  openshift-image-registry | grep image-registry | awk '{print $2}')",
+            "cpd-linux adm -r ${local.installerhome}/repo.yaml -a watson-discovery -n ${self.triggers.namespace} --accept-all-licenses --apply",
+            "cpd-linux -r ${local.installerhome}/repo.yaml -a watson-discovery -n ${self.triggers.namespace} -c ${local.watson-discovery-storageclass} --transfer-image-to $image_registry_route/${self.triggers.namespace} --target-registry-username=kubeadmin --target-registry-password=$(oc whoami -t) --cluster-pull-prefix image-registry.openshift-image-registry.svc:5000/${self.triggers.namespace} --accept-all-licenses --override ${local.installerhome}/watson-discovery-override.yaml --insecure-skip-tls-verify --verbose"
```

Comment:
Use service account token, like the other services do, rather than "oc whoami -t" (it expires)